solr/core/src/java/org/apache/solr/cli/CLIUtils.java

```diff
@@ -321,16 +320,15 @@ public static boolean safeCheckCoreExists(String solrUrl, String coreName, Strin
           final int clamPeriodForStatusPollMs = 1000;
           Thread.sleep(clamPeriodForStatusPollMs);
         }
-        NamedList<Object> existsCheckResult =
-            CoreAdminRequest.getStatus(coreName, solrClient).getResponse();
-        NamedList<Object> status = (NamedList<Object>) existsCheckResult.get("status");
-        NamedList<Object> coreStatus = (NamedList<Object>) status.get(coreName);
-        Map<String, Object> failureStatus =
-            (Map<String, Object>) existsCheckResult.get("initFailures");
-        String errorMsg = (String) failureStatus.get(coreName);
-        final boolean hasName = coreStatus != null && coreStatus.get(NAME) != null;
-        exists = hasName || errorMsg != null;
-        wait = hasName && errorMsg == null && "true".equals(coreStatus.get("isLoading"));
+        final var coreStatusReq = new CoresApi.GetCoreStatus(coreName);
+        final var coreStatusRsp = coreStatusReq.process(solrClient).getParsed();
+        final var coreStatusByName = coreStatusRsp.status;
+        final var coreStatus = coreStatusByName.get(coreName);
+        final var failureStatus = coreStatusRsp.initFailures;
+        final var initFailureForCore = failureStatus.get(coreName);
+        final boolean hasName = coreStatus != null && coreStatus.name != null;
+        exists = hasName || initFailureForCore != null;
+        wait = hasName && initFailureForCore == null && "true".equals(coreStatus.isLoading);
```

Comment:
boo that we are string comparing versus using a boolean...

Reply:
I feel like I see boolean `true` used elsewhere for `isLoading`?

Reply:
Done